webui/react/src/components/PaginatedNotesCard.tsx

```diff
@@ -0,0 +1,203 @@
+    const currentPages = notes.length;
+    onNewPage();
+    handleSwitchPage(currentPages);
+  }, [ notes.length, onNewPage, handleSwitchPage ]);
+
+  const handleSave = useCallback(async (editedNotes: string) => {
+    setEditedContents(editedNotes);
+    await onSave(notes.map((note, idx) => {
+      if (idx === currentPage) {
+        return { contents: editedNotes, name: editedName } as Note;
+      }
+      return note;
+    }));
+  }, [ currentPage, editedName, notes, onSave ]);
+
+  const handleSaveTitle = useCallback(async (newName: string) => {
+    setEditedName(newName);
+    await onSave(notes.map((note, idx) => {
+      if (idx === currentPage) {
+        return { contents: editedContents ?? note?.contents, name: newName } as Note;
+      }
+      return note;
+    }));
+  }, [ currentPage, notes, onSave, editedContents ]);
+
+  const handleDeletePage = useCallback((deletePageNumber: number) => {
+    onDelete(deletePageNumber);
+    setDeleteTarget(deletePageNumber);
+  }, [ onDelete, setDeleteTarget ]);
+
+  const handleEditedNotes = useCallback((newContents: string) => {
+    setEditedContents(newContents);
+  }, []);
+
+  useEffect(() => {
+    if (notes.length === 0) return;
+    if (currentPage < 0) { setCurrentPage(0); fireNoteChangeSignal(); }
+    if (currentPage >= notes.length) { setCurrentPage(notes.length - 1); fireNoteChangeSignal(); }
+  }, [ currentPage, notes.length, fireNoteChangeSignal ]);
+
+  useEffect(() => {
+    setEditedContents(notes?.[currentPage]?.contents ?? '');
+    setEditedName(notes?.[currentPage]?.name ?? '');
+  }, [ currentPage, notes ]);
+
+  const ActionMenu = useCallback((pageNumber: number) => {
+    return (
+      <Menu onClick={({ domEvent }) => domEvent.stopPropagation()}>
+        <Menu.Item
+          danger
+          key="delete"
+          onClick={() => handleDeletePage(pageNumber)}>
+          Delete...
+        </Menu.Item>
+      </Menu>
+    );
+  }, [ handleDeletePage ]);
+
+  if (notes.length === 0) {
+    return (
+      <div className={css.emptyBase}>
+        <div className={css.messageContainer}>
+          <Icon name="document" size="mega" />
+          <p>No notes for this project</p>
+          <Button onClick={handleNewPage}>+ New Page</Button>
+        </div>
+      </div>
+    );
+  }
+
+  return (
+    <div className={css.base}>
+      {notes.length > 0 && (
+        <div className={css.sidebar}>
+          <ul className={css.listContainer} role="list">
+            {(notes as Note[]).map((note, idx) => (
+              <Dropdown
+                disabled={disabled}
+                key={idx}
+                overlay={() => ActionMenu(idx)}
+                trigger={[ 'contextMenu' ]}>
+                <li
+                  className={css.listItem}
+                  style={{
+                    borderColor: idx === currentPage ?
+                      'var(--theme-colors-monochrome-12)' :
+                      undefined,
+                  }}
```

Comment:
**suggestion:** apply it as a dynamic class instead of inline style (e.g. `css.listItem` and `css.listItem css.currentListItem` or something like that)